### PR TITLE
[SPARK-19083]sbin/start-history-server.sh script use of $@ without quotes

### PR DESCRIPTION
**Pull request by zuotingbing**
JIRA Issue: https://issues.apache.org/jira/browse/SPARK-19083#

sbin/start-history-server.sh script use of $@ without quotes, this will affect the length of args which used in HistoryServerArguments::parse(args: List[String])

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala [hunk 1]
```diff
@@ -445,12 +445,24 @@ private[deploy] class Worker(
           // Create local dirs for the executor. These are passed to the executor via the
           // SPARK_EXECUTOR_DIRS environment variable, and deleted by the Worker when the
           // application finishes.
-          val appLocalDirs = appDirectories.getOrElse(appId,
-            Utils.getOrCreateLocalRootDirs(conf).map { dir =>
-              val appDir = Utils.createDirectory(dir, namePrefix = "executor")
-              Utils.chmod700(appDir)
-              appDir.getAbsolutePath()
-            }.toSeq)
+          val appLocalDirs = appDirectories.getOrElse(appId, {
```
Comment (jerryshao) [inline core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala]: What is the purpose of change here, from my understanding it is not related to the issue you mentioned in PR description.

Comment (zuotingbing) [inline core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala]: Sorry, this change is for another issue. Should i close this PR and checkout a new branch for this issue ?

Comment (jerryshao) [inline core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala]: you could push another commit to this branch to revert this change.

**Comment by jerryshao** [comment]
Would you mind checking all the scripts under `bin` and `sbin` to see if this issue also existed in other scripts.

**Comment by zuotingbing** [comment]
@jerryshao Thank you for your advice. I checked off all the scripts under bin and sbin. It seems ok.

**Comment by srowen** [comment]
This is the only instance I see, yes. Other instances are in dev scripts which don't matter.

Could you explain what difference it makes? I read the bash reference and don't fully understand. Would it cause each individual script arg to act as if quoted?

**Comment by vanzin** [comment]
> Would it cause each individual script arg to act as if quoted?

Yes.

LGTM. Merging to master / 2.1.
